### PR TITLE
[ERC20 Launchpad] - Endpoint integration for Community information step

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/9208

## Description of Changes
Added endpoint integration for Community information step

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_TOKENIZED_COMMUNITY`
- Visit `/createTokenCommunity`
- Open `/packages/commonwealth/client/scripts/views/pages/LaunchToken/useCreateCommunity.ts` and add your currently logged-in Ethereum wallet address for the `setSelectedAddress` state - this is only temporary for testing - the actual address selection happens in step 1 of this flow
- Verify the form is prefilled with data constructed from sample token information
- Submit the form and verify that a new token community has been created.

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/CommunityInformationStep/CommunityInformationStep.tsx [hunk 1]
```diff
@@ -1,39 +1,102 @@
...
-    handleContinue();
+  const {
+    mutateAsync: createCommunityMutation,
+    isLoading: createCommunityLoading,
+  } = useCreateCommunityMutation();
+
+  const handleSubmit = async (
+    values: CommunityInformationFormSubmitValues & { communityId: string },
+  ) => {
+    // this condition will never be fulfilled but adding this to avoid typescript errors
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/CommunityInformationStep/CommunityInformationStep.tsx:63]: yeah good stuff, I had to do the same

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/CommunityInformationStep/CommunityInformationStep.tsx [hunk 2]
```diff
@@ -1,39 +1,102 @@
...
+    const baseNode = nodes?.find(
+      (n) => n.ethChainId === commonProtocol.ValidChains.Base,
+    );
+    if (!baseNode || !baseNode.ethChainId) {
+      notifyError('Could not find base chain node');
+      return;
+    }
+
+    try {
+      // call endpoint
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/CommunityInformationStep/CommunityInformationStep.tsx:80]: those comments are redundant imo

**Review by masvelio — APPROVED** [review]
1. image
I remember we had some error displaying under the input for occupied community names 🤔

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/CommunityInformationStep/CommunityInformationStep.tsx (hunk 2 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/CommunityInformationStep/CommunityInformationStep.tsx:80]: removed

**Comment by mzparacha** [comment]
Re: https://github.com/hicommonwealth/commonwealth/pull/9230#pullrequestreview-2309683140

1. Yes, that was removed during the de-coupling of communities config from the app state variable. Basically previously we had all community names in FE so validation was quicker but now we rely on the API to confirm that.

**Review by Israellund — APPROVED** [review]
lgtm

**Pull request merged**